idaes/generic_models/unit_models/surrogate_model.py

```diff
@@ -0,0 +1,135 @@
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Generic template for a surrogate unit model.
+"""
+
+from pyomo.environ import Reference, SolverFactory
+from pyomo.network import Port
+from pyomo.common.config import ConfigValue, In
+
+from idaes.core.process_base import declare_process_block_class, \
+    ProcessBlockData
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+
+__author__ = "Jaffer Ghouse"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("SurrogateModel")
+class SurrogateModelData(ProcessBlockData):
+    """
+    This is the class for a surrogate unit model. This will provide a user
+    with a generic skeleton to add a surrogate unit model.
+    """
+
+    # Create Class ConfigBlock
+    CONFIG = ProcessBlockData.CONFIG()
+    CONFIG.declare("dynamic", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Dynamic model flag",
+        doc="""Surrogate model assumed to be steady-state."""))
+
+    def build(self):
+        """
+        General build method for SurrogateModelBlockData.
+
+        Inheriting models should call `super().build`.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        super(SurrogateModelData, self).build()
+
+    def add_ports(self, name=None, member_list=None, doc=None):
+        """
+        This is a method to build Port objects in a surrogate model and
+        populate this with appropriate port members as specified. User can add
+        as many inlet and outlet ports as required.
+
+        Keyword Args:
+            name : name to use for Port object.
+            dict : dictionary containing variables to be added to the port
+            doc : doc string for Port object
+
+        Returns:
+            A Pyomo Port object and associated components.
+        """
+        # Validate that member_list is a dict
+        if not isinstance(member_list, dict):
+            raise ConfigurationError(
+                "member_list should be a dictionary "
+                "with the keys being the name assigned(strings) and "
+                "values being the variable objects declared in the model.")
+
+        # Create empty Port
+        p = Port(noruleinit=True, doc=doc)
+        setattr(self, name, p)
+
+        # Create References for port members
+        for k in member_list.keys():
+            if not member_list[k].is_indexed():
+                local_name = member_list[k]
+            else:
+                local_name = member_list[k]
+
+            # Add Reference to Port
+            p.add(Reference(local_name), k)
+
+    def initialize(self, custom_initialize=None, outlvl=idaeslog.NOTSET,
+                   solver='ipopt', optarg={'tol': 1e-6}):
```

Comment:
Probably go with optarg=None as default.  @andrewlee94's get_solver() method will get a solver with the default tolerance of 1e-6.  I think it's the new standard in initialization.

Reply:
On that line, the default value for `solver` is now `None` as well, as `get_solver()` will return the configured default solver in this case.